Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -839,6 +839,27 @@ extension RouteController: Router {
 extension RouteController: InternalRouter { }
 
 extension RouteController: ReroutingControllerDelegate {
+    func rerouteControllerWantsSwitchToAlternative(_ rerouteController: RerouteController, response: RouteResponse, options: RouteOptions) {
+        guard let newMainRoute = response.routes?.first else {
+            return
+        }
+        delegate?.router(self,
+                         willTakeAlternativeRoute: newMainRoute,
+                         at: location)
+        updateRoute(with: IndexedRouteResponse(routeResponse: response,
+                                               routeIndex: 0),
```

Comment:
Are you sure that `onSwitchToAlternative` returns route at index `0`? If not, update the guard too (`response.routes?.first`)

Reply:
It will return a route response with selected alternative as first (and only) route

Reply:
Added explicit route index setup from the original RouteInterface